include/aspect/assembly.h

```diff
@@ -428,6 +474,65 @@ namespace aspect
                                       internal::Assembly::CopyData::StokesSystem<dim>      &)> local_assemble_stokes_system_on_boundary_face;
 
         /**
+         * A signal that is called from Simulator::local_assemble_advection_system()
+         * and whose slots are supposed to assemble terms that together form the
+         * Advection system matrix and right hand side.
+         *
+         * The arguments to the slots are as follows:
+         * - The cell on which we currently assemble.
+         * - The advection field that is currently assembled.
+         * - The artificial viscosity computed for the current cell
+         * - The scratch object in which temporary data is stored that
+         *   assemblers may need.
+         * - The copy object into which assemblers add up their contributions.
+         */
+        boost::signals2::signal<void (const typename DoFHandler<dim>::active_cell_iterator &,
+                                      const typename Simulator<dim>::AdvectionField &,
+                                      const double,
+                                      internal::Assembly::Scratch::AdvectionSystem<dim>       &,
+                                      internal::Assembly::CopyData::AdvectionSystem<dim>      &)> local_assemble_advection_system;
+
+        /**
+         * A signal that is called from Simulator::local_assemble_advection_system()
+         * and whose slots are supposed to compute an advection system residual
+         * according to the terms their assemblers implement. If more than one
+         * assembler is connected the residuals are simply added up by the
+         * ResidualWeightSum structure.
+         *
+         * The arguments to the slots are as follows:
```

Comment:
can you mention the return value (``std::vector`` with a value in each quadrature point) here?